optaplanner-wb-screens/optaplanner-wb-domain-editor/optaplanner-wb-domain-editor-client/src/main/java/org/optaplanner/workbench/screens/domaineditor/client/handlers/planner/PlannerDomainHandler.java

```diff
@@ -78,193 +79,180 @@ public void postCommandProcessing( DataModelCommand command ) {
                             || ( (DataObjectChangeEvent) event ).getChangeType() == ChangeType.PACKAGE_NAME_CHANGE
                             || ( (DataObjectChangeEvent) event ).getChangeType() == ChangeType.CLASS_NAME_CHANGE ) {
 
-                        Annotation planningEntityAnnotation = dataObject.getAnnotation( PlannerDomainAnnotations.PLANNING_ENTITY_ANNOTATION );
-                        planningEntityAnnotation.setValue( "difficultyComparatorClass", dataObject.getName() + "." + dataObject.getName() + "Comparator.class" );
-
-                        plannerDataObjectEditorService.call( new RemoteCallback<ComparatorObject>() {
-                            @Override
-                            public void callback( ComparatorObject updatedComparatorObject ) {
-                                dataObject.getNestedClasses().clear();
-                                if ( updatedComparatorObject != null ) {
-                                    dataObject.addNestedClass( updatedComparatorObject );
-
-                                    comparatorDefinition.setValue( "fieldPaths", getAnnotationDef( updatedComparatorObject.getObjectPropertyPathList() ) );
-                                } else {
-                                    dataObject.removeAnnotation( comparatorDefinition.getClassName() );
-                                    Annotation planningEntityAnnotation = dataObject.getAnnotation( PlannerDomainAnnotations.PLANNING_ENTITY_ANNOTATION );
-                                    if ( planningEntityAnnotation != null ) {
-                                        planningEntityAnnotation.removeValue( "difficultyComparatorClass" );
+                        if ( ( (DataObjectChangeEvent) event ).getChangeType() == ChangeType.CLASS_NAME_CHANGE ) {
+                            dataModelerService.call( new RemoteCallback<DataObject>() {
+                                @Override
+                                public void callback( DataObject dataObject ) {
+                                    if ( dataObject != null ) {
+                                        comparatorObjectHandler.updateComparatorObject( dataObject, comparatorObject );
+
+                                        dataObject.getNestedClasses().clear();
+                                        dataObject.addNestedClass( comparatorObject );
+
+                                        Annotation planningEntityAnnotation = dataObject.getAnnotation( PlannerDomainAnnotations.PLANNING_ENTITY_ANNOTATION );
+                                        planningEntityAnnotation.setValue( "difficultyComparatorClass", dataObject.getName() + "." + dataObject.getName() + "Comparator.class" );
+
+                                        dataModelerService.call().saveSource( event.getSource(), event.getPath(), dataObject, null, "Update comparator object after data object class name change" );
```

Comment:
@wmedvede The decision was based on the fact that I couldn't find any occurrences of CLASS_NAME_CHANGE in the project. I reused the change type in the event fired when the file (using 'Rename') was renamed https://github.com/droolsjbpm/kie-wb-common/pull/466/files#diff-5166e0a37add3f6506d1c15648308d73R539. Perhaps I can introduce separate event here, e.g. `DataObjectRenamedEvent` if that makes more sense?. The reason the event is needed is that the handler needs to react appropriately to name changes (e.g. rename the nested comparator class, change parameter types).